blogs/secondary_interface/secondary_interface.md

```diff
@@ -0,0 +1,168 @@
+
+Network observability for secondary interfaces with Multus and SR-IOV plugins
+in Kubernetes can be a complex task, but it's crucial for monitoring and
+troubleshooting network issues in a Kubernetes cluster.
+
+## Overview of how you can achieve network observability for secondary interface
+
+<p align="center">
+  <img src="images/sriov.png" alt="logo" width="75%"/>
+</p>
+
+1. **Multus CNI Plugin**: Multus is a CNI (Container Network Interface) plugin
+    for Kubernetes that allows you to attach multiple network interfaces to pods.
+    In Openshift Multus is used to attach SR-IOV vfs to your pods.
+    For reference and more details about Multus CNI pls refer to
+    https://docs.openshift.com/container-platform/4.13/networking/multiple_networks/understanding-multiple-networks.html
+
+2. **SR-IOV Plugin**: SR-IOV (Single Root I/O Virtualization) is a technology that
+   enables the partitioning of a single PCIe network adapter into multiple virtual
+   functions (VFs).
+   Pods can then use these VFs as secondary network interfaces, achieving higher
+   network performance and isolation.
+   For reference and more details about SR-IOV pls refer to
+   https://docs.openshift.com/container-platform/4.13/networking/hardware_networks/configuring-sriov-device.html
+
+## Network observability eBPF agent enhancements to support the secondary interface
+
+To ensure network observability for secondary interfaces in this setup, and make
+the eBPF agent network namespace aware, eBPF agents need to implement the
+following steps:
+
+1. Using fsNotify Package:
+   Utilize the fsNotify package to be notified when new network namespaces are created.
+   This allows the eBPF agent to keep track of network namespace creation events.
+
+2. Using netlink Package:
+   Employ the netlink package to register when the network interfaces are created or deleted
+   within each network namespace. This will enable the eBPF agent to monitor the interface
+   changes on a per-namespace basis.
+
+3. Attach/Detach eBPF TC Hooks:
+   Add support to the eBPF agent to attach and detach eBPF Traffic Control (TC) hook
+   for network interfaces in non-default network namespaces. This step is crucial for
+   monitoring and controlling network traffic within these network namespaces.
+
+## Configuring SR-IOV objects
+
+1. Install the SR-IOV operator in the environment.
+2. Identify the SR-IOV capable device on the node.
+3. Label the node which has SR-IOV interface with the label
+  `feature.node.kubernetes.io/network-sriov.capable=true`
+4. Create the `SriovNetworkNodePolicy` object
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: mypolicy
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: netdeviceresource
+  nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+  priority: 99
+  numVfs: 50
+  nicSelector:
+    pfNames: ["ens7f0np0#25-49"]
+  deviceType: netdevice
```

Comment:
Should we go in details for each of these fields as we do in OCP doc with bullet points <1> <2> ... ?

Reply:
I don't think so as the scope of this blog is to show how to use netobserv for sriov multus i added the ocp doc that has all details and context